optaplanner-core/src/main/java/org/optaplanner/core/api/domain/variable/IndexShadowVariable.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.api.domain.variable;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.optaplanner.core.api.domain.entity.PlanningEntity;
+import org.optaplanner.core.api.solver.Solver;
+import org.optaplanner.core.impl.heuristic.selector.move.generic.list.ListChangeMove;
+import org.optaplanner.core.impl.heuristic.selector.move.generic.list.ListSwapMove;
+
+/**
+ * Specifies that a bean property (or a field) is an index of this planning value in another entity's
+ * {@link PlanningCollectionVariable}.
+ * <p>
+ * <strong>The source variable must be a disjoint list.</strong> Disjoint means that a planning value is a member of
```

Comment:
This PlanningCollectionVariable needs to be a DISTJOINT_LIST (which is the default). (not strong)